### ISSUE TITLE
The FLOPS per GPU reported for the Megatron GPT model by the DeepSpeed Flops Profiler is much lower than that reported in the logs when we run pretrain_gpt.py[BUG]

### ISSUE THREAD

**Issue by shrutiramesh1988** [microsoft/DeepSpeed#2565]
**Describe the bug**
The FLOPS per GPU reported for the Megatron GPT model by the DeepSpeed Flops Profiler is much lower than that reported in the logs when we run pretrain_gpt.py (of Megatron-DeepSpeed)

Also, when ds_pipeline_enabled=True, the Profiler doesn't generate the Profile Summary. Why does this happen?

**To Reproduce**

**Expected behavior**
The TFLOPS per GPU reported in the profiler Summary should match with that reported by the logs.
Also, the Profile Summary should get printed when ds_pipeline_enabled=True

**ds_report output**
Profiler reports FLOPS per GPU as 13.36 TFLOPS, whereas the log prints the FLOPS per GPU as 125.18 TFLOPs
Profiler printed Samples/s is 49.55 and that printed by the log is 50.510, which match.
Is the FLOPS per GPU computation by the profiler incorrect?

0: -------------------------- DeepSpeed Flops Profiler --------------------------
0: Profile Summary at step 2:
0: Notations:
0: data parallel size (dp_size), model parallel size(mp_size),
0: number of parameters (params), number of multiply-accumulate operations(MACs),
0: number of floating-point operations (flops), floating-point operations per second (FLOPS),
0: fwd latency (forward propagation latency), bwd latency (backward propagation latency),
0: step (weights update latency), iter latency (sum of fwd, bwd and step latency)
0:
0: world size: 128
0: data parallel size: 16
0: model parallel size: 8
0: batch size per GPU: 8
0: params per gpu: 2318.53 M
0: params of model = params per GPU * mp_size: 18548.24 M
0: fwd MACs per GPU: 45999.1 GMACs
0: fwd flops per GPU: 91998.2 G
0: fwd flops of model = fwd flops per GPU * mp_size: 735985.6 G
0: fwd latency: 5.7 s
0: fwd FLOPS per GPU = fwd flops per GPU / fwd latency: 16.15 TFLOPS
0: bwd latency: 14.55 s
0: bwd FLOPS per GPU = 2 * fwd flops per GPU / bwd latency: 12.64 TFLOPS
0: fwd+bwd FLOPS per GPU = 3 * fwd flops per GPU / (fwd+bwd latency): 13.63 TFLOPS
0: step latency: 413.55 ms
0: iter latency: 20.66 s
0: **FLOPS per GPU = 3 * fwd flops per GPU / iter latency: 13.36 TFLOPS**
0: **samples/second: 49.55**
0:

**Logs generated when pretrain_gpt.py is run:**
iteration 50/ 50 | consumed samples: 51200 | consumed tokens: 104857600 | elapsed time per iteration (ms): 20273.0 | learning rate: 9.524E-05 | global batch size: 1024 | lm loss: 7.945235E+00 | loss scale: 4096.0 | number of skipped iterations: 0 | number of nan iterations: 0 | **samples per second: 50.510 | TFLOPs: 125.18** |
time (ms) | forward-compute: 5673.73 | backward-compute: 14185.10 | backward-embedding-all-reduce: 0.01 | optimizer: 412.54 | batch-generator: 10.72

**System info**
 - OS: [Red Hat Enterprise Linux release 8.5]
 - GPU count and types [128 A100 GPUs]
 - Interconnects [ machines are connected with 100 Gbps IB]

**Launcher context**
Using pyxis and slurm to launch

**Docker context**
nvcr.io/nvidia/pytorch:22.10-py3 with pip installed deepspeed

**Additional context**
Running GPT with Megatron-DeepSpeed

**Comment by cli99** [comment]
Hi @shrutiramesh1988, the flops profiler calculates "per GPU" FLOPS whereas the log shows the "per iteration" FLOPs, the difference comes from the mp size which is 8 in your script. the difference between 13.36*8 =106.88 and 125.18 is because "125.18" uses a rough estimation (https://github.com/microsoft/Megatron-DeepSpeed/blob/main/megatron/utils.py#L222-L246).

**Comment by cli99** [comment]
@shrutiramesh1988 currently the flops profiler is not integrated and enabled when `ds_pipeline_enabled=True`. We are adding it soon. thanks.

**Comment by shrutiramesh1988** [comment]
Hello, as per the earlier intimation, we were informed that flops profiler has not yet been integrated and enabled when ds_pipeline_enabled=True. Kindly let us know if this feature has been added now.

**Comment by Jiminator** [comment]
@cli99 Is there any update on whether the flops profiler can be enabled when `ds_pipeline_enabled=True`
